### PR TITLE
cr1051008 restructure freq counter reg map in UCS

### PR DESCRIPTION
**Pull request by houlz0507**
also add the debug code to manually switch cmc image

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c [hunk 1]
```diff
@@ -631,13 +632,16 @@ static unsigned int icap_get_clock_frequency_counter_khz(const struct icap *icap
...
-				if (status == 0x2)
+				if ((status & 0xffff) == 0x2)
 					break;
 				mdelay(1);
 				times--;
 			};
-			freq = reg_rd(icap->icap_clock_freq_counters[idx] +
-				OCL_CLK_FREQ_COUNTER_OFFSET);
+			if ((status & 0xffff) == 0x2) {
+				freq = (status & 0x10000) ?
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c]: Could you define some meaningful macros for bits in status registers for "0x2" and "0x10000"?

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xmc.c [hunk 2]
```diff
@@ -1621,8 +1621,96 @@ static struct attribute_group xmc_attr_group = {
...
+		if (!tmp_buf) {
+			vfree(*image);
+			*image = NULL;
+			return 0;
+		}
+		memcpy(tmp_buf, *image, sz);
+		vfree(*image);
+		sz = total;
+	} else
+		tmp_buf = *image;
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/xmc.c]: please wrap this line w/ "{}" to pass check_patch.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c (hunk 1 above)
Comment (houlz0507) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c]: done

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/xmc.c (hunk 2 above)
Comment (houlz0507) [reply src/runtime_src/core/pcie/driver/linux/xocl/subdev/xmc.c]: done

**Pull request merged**